packages/api/src/services/repository/repository-storage.service.ts

```diff
@@ -54,8 +54,11 @@ export class RepositoryStorageService extends LocalStorageService {
     this.remove(this.SELECTED_REPOSITORY);
   }
 
-  private serializeRepositoryReference(repositoryReference: RepositoryReference): string {
-    return JSON.stringify(repositoryReference);
+  private serializeRepositoryReference(repositoryReference: RepositoryReference): string | null {
+    if (repositoryReference) {
+      return `{"id": "${repositoryReference.id}", "location": "${repositoryReference.location}"}`;
```

Comment:
Two things here:
* Assembling JSON as a string like this is a bad practive. It's not typesafe.
* As far as I can see, the RepositoryReference interface declares only `id` and `location` fields. Why this logic was even needed. If the old code `JSON.stringify(repositoryReference)`  doesn't work here, then the problem is somewhere else most likely.

Reply:
Fixed